### ISSUE TITLE
Improve Workflow for Release Candidate Builds

### ISSUE THREAD

**Issue by CharlieMc0** [zeta-chain/node#1175]
When a release candidate tag is created we should run some testing, build the binaries, and publish a pre-release in GitHub 

This task covers the Create RC Tag > Release Canidate Testing  portion of the flowchart.

![Image](https://github.com/zeta-chain/node/assets/31941002/a7f4ac32-987a-4d1d-a745-fbc18c3290c2)

AC

- Must be triggered automatically when tag matches *-rc pattern
- Release must be marked as "Pre-Release"
- Testing must complete before a release is published 
- The tests themselves  are out of scope for this task since all the tests currently lives at the PR level

**Comment by CharlieMc0** [comment]
This has been started here - https://github.com/zeta-chain/node/pull/1161

**Comment by lumtis** [comment]
Closing as it seems completed
